surfactant/infoextractors/js_file.py

```diff
@@ -26,10 +28,14 @@ def extract_file_info(sbom: SBOM, software: Software, filename: str, filetype: s
 
 def extract_js_info(filename: str) -> object:
     js_info: Dict[str, Any] = {"jsLibraries": []}
-    js_lib_file = pathlib.Path(__file__).parent / "js_library_patterns.json"
+    js_lib_file = (
+        ConfigManager().get_data_dir_path() / "infoextractors" / "js_library_patterns.json"
+    )
 
     # Load expressions from retire.js, should move this file elsewhere
     try:
+        path = ConfigManager().get_data_dir_path() / "infoextractors"
+        path.mkdir(parents=True, exist_ok=True)
         with open(js_lib_file, "r") as regex:
             database = json.load(regex)
     except FileNotFoundError:
```

Comment:
I think this block of code for loading the js_lib_file is what should go into the load_db function.

This block of code could then be replaced by a check:
```python
if js_lib_database is None:
    return None
```

And then either `database = js_lib_database` or update the names of the subsequent uses of `database` => `js_lib_database`